test/event-log.c

```diff
@@ -0,0 +1,299 @@
+#include <context.h>
+
+typedef struct {
+	gchar *tmpdir;
+} EventLogFixture;
+
+typedef struct {
+	goffset maxsize;
+	gint rotates_at;
+} RotationTestConfig;
+
+static void event_log_fixture_set_up(EventLogFixture *fixture,
+		gconstpointer user_data)
+{
+	fixture->tmpdir = g_dir_make_tmp("rauc-event_log-XXXXXX", NULL);
+	g_assert_nonnull(fixture->tmpdir);
+
+	r_context_conf()->configpath = g_strdup("test/test.conf");
+	r_context();
+}
+
+static void config_file_fixture_tear_down(EventLogFixture *fixture,
+		gconstpointer user_data)
+{
+	g_assert_true(rm_tree(fixture->tmpdir, NULL));
+	g_free(fixture->tmpdir);
+	r_context_clean();
+}
+
+/* Test setting up a logger */
+static void event_log_test_setup_logger(EventLogFixture *fixture,
+		gconstpointer user_data)
+{
+	g_autoptr(REventLogger) logger = NULL;
+
+	logger = g_new0(REventLogger, 1);
+	logger->name = g_strdup("testlogger");
+	logger->filename = g_build_filename(fixture->tmpdir, "testfile.log", NULL);
+
+	r_event_log_setup_logger(logger);
+
+	g_assert_true(logger->configured);
+	g_assert_cmpint(logger->filesize, ==, 0);
+}
+
+/* Test setting up a readable (default) logger and writing a simple log message to the logfile */
+static void event_log_test_log_write_simple(EventLogFixture *fixture,
+		gconstpointer user_data)
+{
+	g_autoptr(REventLogger) logger = NULL;
+	GLogField fields[] = {
+		{"MESSAGE", "This is a test (mark) log message", -1 },
+		{"MESSAGE_ID", "1d1b7a5a-a908-4c3a-9004-650c9d2ce850", -1 },
+		{"GLIB_DOMAIN", R_EVENT_LOG_DOMAIN, -1},
+		{"RAUC_EVENT_TYPE", "mark", -1},
+		{"RAUC_SLOT", "rootfs.0", -1},
+		{"BUNDLE_HASH", "b970468f-89e4-4793-9904-06c922902b25", -1},
+		{"SLOT_BOOTNAME", "A", -1},
+	};
+	g_autofree gchar *contents = NULL;
+
+	logger = g_new0(REventLogger, 1);
+	logger->name = g_strdup("testlogger");
+	logger->filename = g_build_filename(fixture->tmpdir, "testfile.log", NULL);
+
+	r_event_log_setup_logger(logger);
+
+	logger->writer(logger, fields, G_N_ELEMENTS(fields));
+
+	g_file_get_contents(logger->filename, &contents, NULL, NULL);
+
+	g_assert_nonnull(strstr(contents, "This is a test (mark) log message"));
+	g_assert_nonnull(strstr(contents, "bundle hash: b970468f-89e4-4793-9904-06c922902b25"));
+}
+
+/* Test setting up a logger with an unwritable file */
+static void event_log_test_log_write_broken(EventLogFixture *fixture,
+		gconstpointer user_data)
+{
+	g_autoptr(REventLogger) logger = NULL;
+	GLogField fields[] = {
+		{"MESSAGE", "This is a test (mark) log message", -1 },
+		{"MESSAGE_ID", "1d1b7a5a-a908-4c3a-9004-650c9d2ce850", -1 },
+		{"GLIB_DOMAIN", R_EVENT_LOG_DOMAIN, -1},
+		{"RAUC_EVENT_TYPE", "mark", -1},
+		{"RAUC_SLOT", "rootfs.0", -1},
+		{"BUNDLE_HASH", "b970468f-89e4-4793-9904-06c922902b25", -1},
+		{"SLOT_BOOTNAME", "A", -1},
+	};
+	g_autofree gchar *contents = NULL;
+
+	logger = g_new0(REventLogger, 1);
+	logger->name = g_strdup("testlogger");
+	logger->filename = g_strdup("/this/path/does/not/exist/testfile.log");
```

Comment:
Another case would be a full device. You can use `/dev/full` for that:
```
# echo foo > /dev/full 
-bash: echo: write error: No space left on device
```

Reply:
Added test case "/event-log/log-writer/no-space-left" for this.